### PR TITLE
add git-secret only if the sshPrivateKeySecretName provided

### PR DESCRIPTION
**Pull request by ajayy004**
## Description

ssh-secret not required when sshPrivateKeySecretName not provided (houston only sends sshPrivateKeySecretName when sshKey config set )

### Related
- https://github.com/astronomer/issues/issues/4059

### CODE REVIEW

File: tests/chart_tests/test_git_sync_relay_deployment.py [hunk 1]
```diff
@@ -121,3 +121,60 @@ def test_gsr_deployment_with_ssh_credentials_and_known_hosts(self, kube_version)
             "--wait=333",
             "--root=/git",
         ]
+
+    def test_gsr_deployment_without_ssh_credentials_and_known_hosts(self, kube_version):
+        """Test that a valid deployment is rendered when enabling git-sync with ssh credentials and known hosts and other custom configs."""
```
Comment (danielhoherd) [inline tests/chart_tests/test_git_sync_relay_deployment.py:126]: I don't think this test is doing what we need it to do. EG: it doesn't test that there is no secret.

Comment (danielhoherd) [reply tests/chart_tests/test_git_sync_relay_deployment.py:126]: Wait, I see that it actually is checking that now.

Comment (danielhoherd) [inline tests/chart_tests/test_git_sync_relay_deployment.py:126]:
```suggestion
        """Test that a valid deployment is rendered when enabling git-sync without ssh credentials."""
```

Comment (ajayy004) [reply tests/chart_tests/test_git_sync_relay_deployment.py]: Fixed

**Pull request merged**